cloudapi/config.go

```diff
@@ -158,6 +162,9 @@ type Config struct {
 
 	// Deprecated: Remove this when migration from the cloud output v1 will be completed
 	MaxMetricSamplesPerPackage null.Int `json:"maxMetricSamplesPerPackage" envconfig:"K6_CLOUD_MAX_METRIC_SAMPLES_PER_PACKAGE"`
+
+	// WarningMessage is a string that will be shown to the user if the config has some invalid values.
+	WarningMessage string `json:"-"`
```

Comment:
Another comment about it, I have the feeling it could be better to have it as a returned value of the _consolidation_ method. I know it will be a breaking change and it will require more work but it adds some better feedback for the code depending on it, making this change safer. Most of the code that uses this use this method should be internal (right?) so we aren't creating breaking changes for the users.
I think you originally thought about it, so probably you have a wider analysis of it.

Reply:
It's not a breaking change. I've considered this and discarded it since I found no value in having a warning message as a peace.

My point about having this as the property is that I don't consider this warning as something you need immediately after the config consolidation. So, it's just a config property. It's so different from the error since we should stop the process if the `GetConsolidatedConfig` errors. But a warning isn't like that. Next, that change requires streaming down this warning along with the config anyway, and this likely wants to improve the code from the reader's perspective.

From your standpoint, you are having this separated. What do you think will be an improvement? And what problem will this separation solve?

Reply:
>And what problem will this separation solve?

I see the following code block implemented in multiple places:
```go
<configVar>.WarningMessage != "" {
```
so if the next time we implement a new part of code where consolidation would be required, or refactoring one of the already existing, it is a piece of logic that could be forgotten and not notified, where instead we could have the compiler doing the work for us and benefit from it.

Reply:
And the reason, behind so much pickiness, is the attempt to reduce adding more things to be aware of, in a part of the code where we already have a bunch of unknown/unwritten logic. As the discussions before for the unknown comments show.

Reply:
>I see the following code block implemented in multiple places:

Maybe I'm missing something, but the only change will be that instead of the 

```
<configVar>.WarningMessage != "" {
```
We will end up with the 
```
warningMessage != "" {
```

And the predominant issue here is that you likely won't use the `warn` in the same method where you execute the `GetConsolidatedConfig`. 

So still, to be honest, I don't see it as a valuable improvement that deserves a decrease in the readability of the code. 

However, I might be wrong. So let's wait for the third opinion here. It's not a big deal to implement this and as I said it's not a breaking change, but rather an internal thing.

Reply:
>Maybe I'm missing something, but the only change will be that instead of the

The main change to me is that when you call `GetConsolidatedConfig` instead of getting `c, err :=` you would get `c, warning/s, err :=`.

>And the predominant issue here is that you likely won't use the warn in the same method where you execute the GetConsolidatedConfig.

I agree that it would be the predominant issue in that case, but It seems to me that the current code does the check exactly after invoking `GetConsolidatedConfig`.

Reply:
Okay, you're right here :+1: Let me adjust

Reply:
Done with the latest force push